### PR TITLE
[RFR][1LP] quadicons: support complex keys

### PR DESCRIPTION
**Pull request by izapolsk**
It should be possible to look for quad icons using several keys not only name. 
In 5.8,  the view is automatically switched to List View since all possible keys are available only in List View.
In 5.9, JS Api provides all neccessary keys, so, it isn't neccessary to change view mode.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit e6f5b868e17028124aefec88cf2482a46fcc4ae5:

`widgetastic_manageiq/__init__.py`:
- :bangbang: Line 2710:19: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 97db1e9335e653fb9e06d966a7f55f9938e64280:

`widgetastic_manageiq/__init__.py`:
- :bangbang: Line 2715:19: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 693983844dd1e565af4df50b653638b95f238fae:

`widgetastic_manageiq/__init__.py`:
- :bangbang: Line 2763:19: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit cb2611da58d66186c8e35db89f8a903880c0b6d1:

`widgetastic_manageiq/__init__.py`:
- :bangbang: Line 2766:19: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit baf626674d9962c18b410748c3ba554e017bc68c:

`widgetastic_manageiq/__init__.py`:
- :bangbang: Line 2770:64: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit bd1c549c414b34c8f5a11a90c079ee9738b4fcb0:

`widgetastic_manageiq/__init__.py`:
- :bangbang: Line 2770:64: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit eaa24791998bf3d584f9c533e0c91d365c06f239:

`widgetastic_manageiq/__init__.py`:
- :bangbang: Line 2770:64: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 2a3cfa0e5fce37645e2b0cb62bbb84471bc50e36:

`widgetastic_manageiq/__init__.py`:
- :red_circle: Line 2916:30: **F821** *undefined name 'ListView'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9732b0bcd12edb9bc9855d40dd31cf787c952281:

`widgetastic_manageiq/__init__.py`:
- :red_circle: Line 2919:30: **F821** *undefined name 'ListView'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c813001e6418255db56ddcf6dc4b6146182017a8:

`widgetastic_manageiq/__init__.py`:
- :red_circle: Line 2926:30: **F821** *undefined name 'ListView'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit eb7df2b87852a6eaa1a8a41a2767b903915550da:

`widgetastic_manageiq/__init__.py`:
- :red_circle: Line 2934:30: **F821** *undefined name 'ListView'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 33691e8418de745d9f6e7658004d020a26895300:

`widgetastic_manageiq/__init__.py`:
- :red_circle: Line 2934:30: **F821** *undefined name 'ListView'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9814869f37a1bd66bc926d4d341da813650d1d7b:

`widgetastic_manageiq/__init__.py`:
- :red_circle: Line 2934:30: **F821** *undefined name 'ListView'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 796446d3ade10e62a4054d261eadc77bab1fb579:

`cfme/containers/image.py`:
- :bangbang: Line 12:63: **E231** *missing whitespace after ','*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by gshefer** [comment]
@izapolsk please run all container tests to verify the results.

**Comment by dajoRH** [comment]
Lint report for commit 6559cfd27937e475d78d4018f445e3bbd20b474d:

`widgetastic_manageiq/__init__.py`:
- :bangbang: Line 2840:13: **E303** *too many blank lines (2)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: widgetastic_manageiq/__init__.py [hunk 1]
```diff
@@ -2666,43 +2689,54 @@ class NonJSBaseEntity(View):
...
-                return item(name=self.name)
+            if item(id=self.id).is_displayed:
+                return item(id=self.id)
         else:
-            raise NoSuchElementException("Item {name} isn't found on page".format(name=self.name))
+            raise NoSuchElementException("Item {id} isn't found on page".format(id=self.id))
 
     def __getattr__(self, name):
-        if name.startswith('__'):
+        if name.startswith('__') or name == 'is_displayed':
```
Comment (RonnyPfannschmidt) [inline widgetastic_manageiq/__init__.py]: that bit looks wrong - is_displayed is a property, getattr shouldn't even care

File: widgetastic_manageiq/__init__.py [hunk 2]
```diff
@@ -2780,62 +2868,49 @@ def get_all(self, surf_pages=False):
...
-            if by_partial_name in entity.name:
-                remaining_entities.append(entity)
-            # todo: by_type and by_regexp will be implemented later if needed
-        if not remaining_entities:
-            raise ItemNotFound("Entities matching to {name} "
-                               "aren't found on this page".format(name=by_partial_name))
-        return remaining_entities
-
-    def get_entity(self, by_name, surf_pages=False):
+    def get_entity(self, surf_pages=False, **keys):
```
Comment (RonnyPfannschmidt) [inline widgetastic_manageiq/__init__.py]: this is a **breaking** api change, we need to correctly handle it - 3rd party users may be taken by complete surprise

Comment (mfalesni) [reply widgetastic_manageiq/__init__.py]: I don't think there are any users apart form us :laughing:

Comment (izapolsk) [reply widgetastic_manageiq/__init__.py]: it won't be difficult to me to add support of api. However, I'm not sure it is necessary.
@psav, what do you think ?

File: widgetastic_manageiq/__init__.py (hunk 1 above)
Comment (izapolsk) [reply widgetastic_manageiq/__init__.py]: you're right. my bad, I forgot to remove it

File: widgetastic_manageiq/__init__.py (hunk 2 above)
Comment (psav) [reply widgetastic_manageiq/__init__.py]: I don't think it's necessary right now

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Pull request merged**